Templates/integration_template.ps1

```diff
@@ -1,85 +1,37 @@
 #>
 
-# TODO: Customize these paramters...
-$DSCModuleName      = 'xNetworking'
-$DSCResourceName    = 'MSFT_x<ResourceName>'
-$RelativeModulePath = "$DSCModuleName.psd1"
+# TODO: Customize these parameters...
+$Global:DSCModuleName      = 'x<ModuleName>' # Example xNetworking
+$Global:DSCResourceName    = 'MSFT_x<ResourceName>' # Example MSFT_xFirewall
 # /TODO
 
 #region HEADER
-# Temp Working Folder - always gets remove on completion
-$WorkingFolder = Join-Path -Path $env:Temp -ChildPath $DSCResourceName
-
-# Copy to Program Files for WMF 4.0 Compatability as it can only find resources in a few known places.
-$moduleRoot = "${env:ProgramFiles}\WindowsPowerShell\Modules\$DSCModuleName"
-
-# If this module already exists in the Modules folder, make a copy of it in
-# the temporary folder so that it isn't accidentally used in this test.
-if(-not (Test-Path -Path $moduleRoot))
+if ( (-not (Test-Path -Path '.\DSCResource.Tests\')) -or `
+     (-not (Test-Path -Path '.\DSCResource.Tests\TestHelper.psm1')) )
 {
-    $null = New-Item -Path $moduleRoot -ItemType Directory
+    & git @('clone','https://github.com/PowerShell/DscResource.Tests.git')
 }
 else
 {
-    # Copy the existing folder out to the temp directory to hold until the end of the run
-    # Delete the folder to remove the old files.
-    $tempLocation = Join-Path -Path $env:Temp -ChildPath $DSCModuleName
-    Copy-Item -Path $moduleRoot -Destination $tempLocation -Recurse -Force
-    Remove-Item -Path $moduleRoot -Recurse -Force
-    $null = New-Item -Path $moduleRoot -ItemType Directory
-}
-
-
-# Copy the module to be tested into the Module Root
-Copy-Item -Path $PSScriptRoot\..\..\* -Destination $moduleRoot -Recurse -Force -Exclude '.git'
-
-# Import the Module
-$Splat = @{
-    Path = $moduleRoot
-    ChildPath = $RelativeModulePath
-    Resolve = $true
-    ErrorAction = 'Stop'
-}
-$DSCModuleFile = Get-Item -Path (Join-Path @Splat)
-
-# Remove all copies of the module from memory so an old one is not used.
-if (Get-Module -Name $DSCModuleFile.BaseName -All)
-{
-    Get-Module -Name $DSCModuleFile.BaseName -All | Remove-Module
-}
-
-# Import the Module to test.
-Import-Module -Name $DSCModuleFile.FullName -Force
-
-<#
-  This is to fix a problem in AppVoyer where we have multiple copies of the resource
-  in two different folders. This should probably be adjusted to be smarter about how
-  it finds the resources.
-#>
-if (($env:PSModulePath).Split(';') -ccontains $pwd.Path)
-{
-    $script:tempPath = $env:PSModulePath
-    $env:PSModulePath = ($env:PSModulePath -split ';' | Where-Object {$_ -ne $pwd.path}) -join ';'
-}
-
-# Preserve and set the execution policy so that the DSC MOF can be created
-$executionPolicy = Get-ExecutionPolicy
-if ($executionPolicy -ne 'Unrestricted')
-{
-    Set-ExecutionPolicy -ExecutionPolicy Unrestricted -Force
-    $rollbackExecution = $true
+    & git @('-C',(Join-Path -Path (Get-Location) -ChildPath '\DSCResource.Tests\'),'pull')
```

Comment:
Is it definitely slowing it down? I thought it would slow the tests down slightly, but when I was watching them it didn't seem to slow it down at all. Of course you can only see this when performing local tests because then you get to see the "Already up-to-date" message from Git.exe.

The big slow down now is the PSScriptAnalyzer tests that now fire because of the changes made to DSCResource.Tests yesterday. Although it's only a minute or two slower usually.